src/containers/SortingView.tsx

```diff
@@ -102,40 +101,31 @@ const SortingView: React.FunctionComponent<Props> = (props) => {
 
   const handleUnitClicked = useCallback((unitId, event) => {
     if (event.ctrlKey) {
-      if (selectedUnitIds[unitId + '']) {
-        setSelectedUnitIds({
-          ...selectedUnitIds,
-          [unitId + '']: false
-        })
+      if (selection.selectedUnitIds.includes(unitId)) {
+        selectionDispatch({type: 'SetSelectedUnitIds', selectedUnitIds: selection.selectedUnitIds.filter(uid => (uid !== unitId))})
       }
       else {
-        setSelectedUnitIds({
-          ...selectedUnitIds,
-          [unitId + '']: true
-        })
-        setFocusedUnitId(unitId)
+        selectionDispatch({type: 'SetSelectedUnitIds', selectedUnitIds: [...selection.selectedUnitIds, unitId]})
       }
     }
-    else if (event.shiftKey) {
-      if (focusedUnitId !== null) {
-        const unitIds = intrange(focusedUnitId, unitId)
-        const newSelectedUnitIds = {...selectedUnitIds}
-        const sortingInfo = sorting?.sortingInfo
-        if (sortingInfo) {
-          for (let uid of unitIds) {
-            if (sortingInfo.unit_ids.includes(uid)) {
-              newSelectedUnitIds[uid] = true
-            }
+    else if ((event.shiftKey) && (anchorUnitId !== null)) {
+      const unitIds = intrange(anchorUnitId, unitId)
+      const newSelectedUnitIds = [...selection.selectedUnitIds]
+      const sortingInfo = sorting?.sortingInfo
+      if (sortingInfo) {
+        for (let uid of unitIds) {
+          if (sortingInfo.unit_ids.includes(uid)) {
+            if (!newSelectedUnitIds.includes(uid)) newSelectedUnitIds.push(uid)
```

Comment:
There's no reason to think we would have non-numeric or non-sorted unit IDs, right? I always assumed that the order they display on the screen would make sense relative to the sequence they appeared in the list, but I suppose this rule could give us problems if the units came in in some arbitrary unsorted order.

Of course we can probably just sort them, anyway. I just had a sudden concern about the edge case.